pkg/kubelet/certificate/certificate_manager.go

```diff
@@ -154,15 +211,47 @@ func (m *manager) Start() {
 	}, 0)
 }
 
+func getCurrentCertificateOrBootstrap(
+	store Store,
+	bootstrapCertificatePEM []byte,
+	bootstrapKeyPEM []byte) (*CertKeyData, bool, error) {
+
+	certKeyData, err := store.Current()
+	if err == nil {
+		return certKeyData, false, nil
+	}
+
+	if _, ok := err.(*NoCertKeyError); !ok {
+		return nil, false, err
+	}
+
+	if bootstrapCertificatePEM == nil || bootstrapKeyPEM == nil {
+		return nil, false, fmt.Errorf("no cert/key available and no bootstrap cert/key to fall back to")
+	}
+
+	cert, err := tls.X509KeyPair(bootstrapCertificatePEM, bootstrapKeyPEM)
+	if err != nil {
+		return nil, false, err
+	}
+	certs, err := x509.ParseCertificates(cert.Certificate[0])
+	if err != nil {
+		return nil, false, fmt.Errorf("unable to parse certificate data: %v", err)
+	}
+	cert.Leaf = certs[0]
```

Comment:
Length of the output will equal the length of the input, so in this case, always 1, as long as there is no err.